content/en/docs/tasks/administer-cluster/memory-manager.md

```diff
@@ -0,0 +1,390 @@
+```
+Nov 18 10:41:22 dell8 kubelet[7227]: I1118 10:41:22.451426    7227 scope_container.go:53] [topologymanager] Best TopologyHint for (pod: gpod_default(2a97a5c6-0577-4956-9f61-153b735e95f6) container: gpod): {01 true}
+```
+
+The best hint (`{01 true}`) indicates where to allocate all the resources. Topology Manager tests this hint against its current policy, and based on the verdict, it either admits the pod to the node or rejects it.  
+
+Also, the grep might be used to search for occurrence of `[memorymanager]` in the logs:
+```
+Nov 03 13:23:00 dell8 kubelet[48363]: I1103 13:23:00.347272   48363 memory_manager.go:204] [memorymanager] Set container "046927bb304a02c1969d6389b28c6f279069e4d91bf6f726e028d714c06db0b1" cpuset.mems to "0"
+```
+
+### State file
+
+Let us first deploy a sample `Guaranteed` pod whose specification is as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: guaranteed
+spec:
+  containers:
+  - name: guaranteed
+    image: consumer
+    imagePullPolicy: Never
+    resources:
+      limits:
+        cpu: "2"
+        memory: 150Gi
+      requests:
+        cpu: "2"
+        memory: 150Gi
+    command: ["sleep","infinity"]
+``` 
+
+Next, let us log into the node where it was deployed and examine the state file in `/var/lib/kubelet/memory_manager_state`:
+```
+{
+   "policyName":"static",
+   "machineState":{
+      "0":{
+         "numberOfAssignments":1,
+         "memoryMap":{
+            "hugepages-1Gi":{
+               "total":0,
+               "systemReserved":0,
+               "allocatable":0,
+               "reserved":0,
+               "free":0
+            },
+            "memory":{
+               "total":134987354112,
+               "systemReserved":3221225472,
+               "allocatable":131766128640,
+               "reserved":131766128640,
+               "free":0
+            }
+         },
+         "nodes":[
+            0,
+            1
+         ]
+      },
+      "1":{
+         "numberOfAssignments":1,
+         "memoryMap":{
+            "hugepages-1Gi":{
+               "total":0,
+               "systemReserved":0,
+               "allocatable":0,
+               "reserved":0,
+               "free":0
+            },
+            "memory":{
+               "total":135286722560,
+               "systemReserved":2252341248,
+               "allocatable":133034381312,
+               "reserved":29295144960,
+               "free":103739236352
+            }
+         },
+         "nodes":[
+            0,
+            1
+         ]
+      }
+   },
+   "entries":{
+      "fa9bdd38-6df9-4cf9-aa67-8c4814da37a8":{
+         "guaranteed":[
+            {
+               "numaAffinity":[
+                  0,
+                  1
+               ],
+               "type":"memory",
+               "size":161061273600
+            }
+         ]
+      }
+   },
+   "checksum":4142013182
+}
+```
+
+It can be deduced from the state file that the pod was affined to both NUMA nodes, i.e.:

```

Comment:
Affined? Perhaps we should explain this term.

Reply:
We decided to rephrase to "pinning" as more suitable term.

Reply:
I added some description:

> Pinned term means that pod's memory consumption is constrained (through `cgroups` configuration) to these NUMA nodes.